src/common/isBigNumber.validator.ts

```diff
@@ -0,0 +1,17 @@
+import { BigNumber } from 'ethers';
+import {
+  ValidatorConstraint,
+  ValidatorConstraintInterface,
+  ValidationArguments,
+} from 'class-validator';
+
+@ValidatorConstraint({ name: 'isBigNumber', async: false })
+export class IsBigNumber implements ValidatorConstraintInterface {
+  validate(val: unknown, args: ValidationArguments) {
+    return val instanceof BigNumber;
```

Comment:
I think we should add more asserts here. Checking for format of the `val` or something